From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Samuel Maddock <smaddock@slack-corp.com>
Date: Thu, 9 Oct 2025 22:07:38 -0400
Subject: chore: disable protocol handler dcheck

https://chromium-review.googlesource.com/c/chromium/src/+/6727594

The above CL introduces a new extensions API to register custom protocol
handlers. A DCHECK causes Electron to crash until we provide our own
registry. This patch disables the check until we support this.

diff --git a/extensions/browser/api/protocol_handlers/protocol_handlers_manager.cc b/extensions/browser/api/protocol_handlers/protocol_handlers_manager.cc
index fe6add818b6ccbdbb699738b5d5d87eb558a6830..e1be9ea485afdae67122fd9e7a293f6527db3598 100644
--- a/extensions/browser/api/protocol_handlers/protocol_handlers_manager.cc
+++ b/extensions/browser/api/protocol_handlers/protocol_handlers_manager.cc
@@ -95,7 +95,6 @@ void ProtocolHandlersManager::OnExtensionLoaded(content::BrowserContext*,
       browser_context_);
   // Can be null for tests using dummy profiles.
   if (!registry) {
-    CHECK_IS_TEST();
     return;
   }
 
@@ -114,7 +113,6 @@ void ProtocolHandlersManager::OnExtensionUnloaded(
       browser_context_);
   // Can be null for tests using dummy profiles.
   if (!registry) {
-    CHECK_IS_TEST();
     return;
   }
 
@@ -135,7 +133,6 @@ void ProtocolHandlersManager::ProtocolHandlersSanityCheck() {
           browser_context_);
   // Can be null for tests using dummy profiles.
   if (!ph_registry) {
-    CHECK_IS_TEST();
     return;
   }
   for (const auto& handler : ph_registry->GetExtensionProtocolHandlers()) {
